### ISSUE TITLE
Can anyone share a train option that produces a good mesh?

### ISSUE THREAD

**Issue by hanjoonwon** [autonomousvision/sdfstudio#299]
![image](https://github.com/autonomousvision/sdfstudio/assets/125524043/d14d6f55-d0b8-4d2c-9bbe-6b5294030b5a)

https://drive.google.com/drive/folders/1lHsiW8MGQcVTrQT9LtJjH8Z6ChMiiVlf?usp=drive_link
https://drive.google.com/drive/folders/1Q9zSM8sCsQ4n-6QKl4GEujuznx-abxvh?usp=drive_link

These are the images I've tried so far

and my train steps

1. ns-process-data images --data

2.python  path/process_nerfstudio_to_sdfstudio.py  --data-type colmap 
--mono-prior  --scene-type object --data path --output-dir path/
 --omnidata-path  sdfstudio/omnidata/omnidata_tools/torch/ --pretrained-models sdfstudio/omnidata/omnidata_tools/torch/pretrained_models/
 
3.ns-train neus-facto --pipeline.datamanager.train-num-rays-per-batch 2048 \
--pipeline.model.sdf-field.use-grid-feature False \
--pipeline.model.sdf-field.inside-outside False \
--pipeline.model.background-model mlp \
--pipeline.model.mono-depth-loss-mult 0.0 \
--pipeline.model.mono-normal-loss-mult 0.1 \
--vis wandb \
--trainer.steps_per_save 5000 \
--trainer.steps-per-eval-image 5000 \
--trainer.max-num-iterations 300000 \

https://github.com/autonomousvision/sdfstudio/blob/370902a10dbef08cb3fe4391bd3ed1e227b5c165/docs/sdfstudio-examples.md?plain=1#L24

I've tried the parameters in this link, and I've also tried other people's options. What could be the problem?
I've tried monosdf, volsdf, and bakedangelo.

**Comment by niujinshuchong** [comment]
Hi, I can get reasonable results with your data. 
<img width="974" alt="image" src="https://github.com/autonomousvision/sdfstudio/assets/13434986/cd086c60-f744-4494-b0f1-aa4b33752ae5">


I used the following command:
```
ns-train neus --pipeline.model.sdf-field.inside-outside False --pipeline.model.sdf-field.bias 0.3 --pipeline.model.sdf-field.beta-init 0.3 --pipeline.model.sdf-field.use-appearance-embedding True --trainer.steps-per-eval-image 500 --pipeline.model.near-plane 0.05 --pipeline.model.far-plane 2. --pipeline.model.overwrite-near-far-plane True --vis wandb nerfstudio-data --data data/owl2-colmap --downscale-factor 2 
```

I also tested with monocular prior.
```
OMP_NUM_THREADS=4 CUDA_VISIBLE_DEVICES=6 ns-train neus --pipeline.model.sdf-field.inside-outside False --pipeline.model.sdf-field.bias 0.3 --pipeline.model.sdf-field.beta-init 0.3 --pipeline.model.sdf-field.use-appearance-embedding True --trainer.steps-per-eval-image 500 --pipeline.model.near-plane 0.05 --pipeline.model.far-plane 2. --pipeline.model.overwrite-near-far-plane True --pipeline.model.mono-normal-loss-mult 0.01 --vis wandb sdfstudio-data --data data/owl2-sdfstudio --include-mono-prior True
```
 The table is reconstructed better with the prior:
![image](https://github.com/autonomousvision/sdfstudio/assets/13434986/c9d39edf-5a0d-4a52-835e-26dea176bd31)

**Comment by hanjoonwon** [comment]
> Hi, I can get reasonable results with your data. <img alt="image" width="974" src="https://private-user-images.githubusercontent.com/13434986/312539723-cd086c60-f744-4494-b0f1-aa4b33752ae5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzODczNzQsIm5iZiI6MTcxMDM4NzA3NCwicGF0aCI6Ii8xMzQzNDk4Ni8zMTI1Mzk3MjMtY2QwODZjNjAtZjc0NC00NDk0LWIwZjEtYWE0YjMzNzUyYWU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDAzMzExNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwNjk3NzM4YjViNjJlMTRkNjgzMDg3OWEyNjczMTFkYTQ5ZTFlYTEwYzE0MmJlYTdmZWJmNjQ4YjFlZjU1YmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5norAbjazGyIBum_oRvu8rXDBFfxsQa9GIEgge-QdBs">
> 
> I used the following command:
> 
> ```
> ns-train neus --pipeline.model.sdf-field.inside-outside False --pipeline.model.sdf-field.bias 0.3 --pipeline.model.sdf-field.beta-init 0.3 --pipeline.model.sdf-field.use-appearance-embedding True --trainer.steps-per-eval-image 500 --pipeline.model.near-plane 0.05 --pipeline.model.far-plane 2. --pipeline.model.overwrite-near-far-plane True --vis wandb nerfstudio-data --data data/owl2-colmap --downscale-factor 2 
> ```
> 
> I also tested with monocular prior.
> 
> ```
> OMP_NUM_THREADS=4 CUDA_VISIBLE_DEVICES=6 ns-train neus --pipeline.model.sdf-field.inside-outside False --pipeline.model.sdf-field.bias 0.3 --pipeline.model.sdf-field.beta-init 0.3 --pipeline.model.sdf-field.use-appearance-embedding True --trainer.steps-per-eval-image 500 --pipeline.model.near-plane 0.05 --pipeline.model.far-plane 2. --pipeline.model.overwrite-near-far-plane True --pipeline.model.mono-normal-loss-mult 0.01 --vis wandb sdfstudio-data --data data/owl2-sdfstudio --include-mono-prior True
> ```
> 
> The table is reconstructed better with the prior: ![image](https://private-user-images.githubusercontent.com/13434986/312540264-c9d39edf-5a0d-4a52-835e-26dea176bd31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzODczNzQsIm5iZiI6MTcxMDM4NzA3NCwicGF0aCI6Ii8xMzQzNDk4Ni8zMTI1NDAyNjQtYzlkMzllZGYtNWEwZC00YTUyLTgzNWUtMjZkZWExNzZiZDMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE0VDAzMzExNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ0MGVjNDQ4MzIwMTliYjE3ZGM4NjA0M2VmMThiMzM2YjFkZDNmOTI0MTRiYzZiMGNiMGYxNDY2MjNlMTg2ZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QaGI48oFDYx4y-44qAKivEzqaJt1MC9CUkF7Dm-rGJM)

@niujinshuchong 
Thank you so much :) In addition to these datasets, I also want to accurately reconstruct battery images and how do I find the right train option to train not only owl images but also other images?

**Comment by hanjoonwon** [comment]
> Hi, I can get reasonable results with your data. <img alt="image" width="974" src="https://private-user-images.githubusercontent.com/13434986/312539723-cd086c60-f744-4494-b0f1-aa4b33752ae5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2NDc3MTIsIm5iZiI6MTcxMDY0NzQxMiwicGF0aCI6Ii8xMzQzNDk4Ni8zMTI1Mzk3MjMtY2QwODZjNjAtZjc0NC00NDk0LWIwZjEtYWE0YjMzNzUyYWU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDAzNTAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmYzliYjdhNzRhNWIzNTY2ODYxZmJmYWJjYzM2MDRiOWE3ZGM2NjMzMGJiMDBjMDZkNTRmMzJmYjVkMjRkMGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.M-arvSOtAFJV6A0pJrHjx0J369yv6-EGe2qI5suOHBc">
> 
> I used the following command:
> 
> ```
> ns-train neus --pipeline.model.sdf-field.inside-outside False --pipeline.model.sdf-field.bias 0.3 --pipeline.model.sdf-field.beta-init 0.3 --pipeline.model.sdf-field.use-appearance-embedding True --trainer.steps-per-eval-image 500 --pipeline.model.near-plane 0.05 --pipeline.model.far-plane 2. --pipeline.model.overwrite-near-far-plane True --vis wandb nerfstudio-data --data data/owl2-colmap --downscale-factor 2 
> ```
> 
> I also tested with monocular prior.
> 
> ```
> OMP_NUM_THREADS=4 CUDA_VISIBLE_DEVICES=6 ns-train neus --pipeline.model.sdf-field.inside-outside False --pipeline.model.sdf-field.bias 0.3 --pipeline.model.sdf-field.beta-init 0.3 --pipeline.model.sdf-field.use-appearance-embedding True --trainer.steps-per-eval-image 500 --pipeline.model.near-plane 0.05 --pipeline.model.far-plane 2. --pipeline.model.overwrite-near-far-plane True --pipeline.model.mono-normal-loss-mult 0.01 --vis wandb sdfstudio-data --data data/owl2-sdfstudio --include-mono-prior True
> ```
> 
> The table is reconstructed better with the prior: ![image](https://private-user-images.githubusercontent.com/13434986/312540264-c9d39edf-5a0d-4a52-835e-26dea176bd31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2NDc3MTIsIm5iZiI6MTcxMDY0NzQxMiwicGF0aCI6Ii8xMzQzNDk4Ni8zMTI1NDAyNjQtYzlkMzllZGYtNWEwZC00YTUyLTgzNWUtMjZkZWExNzZiZDMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDAzNTAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiMmJkMDFjMTM0ZjBkMDU5ODJhOWU5NWRlZDU4NGI2MDg1NGFmYzU0OTkzY2FhM2IyOGVmMmIzYmZlYmUyYmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.RPocRt6tu1N4VanMsYLSVZHg-MB6VzH5oo3sBZTrgG0)

@niujinshuchong thank you so much :) 
I tried applying this train option to the battery image as is, but with poor quality results. How can I find the right parameters to get good results for other images?
